python/lsst/fgcmcal/utilities.py

```diff
@@ -394,30 +398,36 @@ def computeCcdOffsets(camera, pixelScale):
                                               ('X_SIZE', 'i4'),
                                               ('Y_SIZE', 'i4')])
 
+    # Generate fake WCSs centered at 180/0 to avoid the RA=0/360 problem,
+    # since we are looking for relative positions
+    boresight = geom.SpherePoint(180.0*geom.degrees, 0.0*geom.degrees)
+    # TODO: DM-17597 will update testdata_jointcal so that the test data
+    # does not have nan as the boresight angle for HSC data.  For the
+    # time being, there is this ungainly hack.
+    if camera.getName() == 'HSC' and np.isnan(defaultOrientation):
+        orientation = 270*geom.degrees
+    else:
+        orientation = defaultOrientation*geom.degrees
+    flipX = False
+
     for i, detector in enumerate(camera):
-        # new version, using proper rotations
-        #  but I worry this only works with HSC, as there's a unit inconsistency
-        camPoint = detector.getCenter(afwCameraGeom.FOCAL_PLANE)
-        pixelSize = detector.getPixelSize()[0]  # Assumes x and y pixel sizes in arcsec are the same
-        camPoint.scale(1.0/pixelSize)
+        ccdOffsets['CCDNUM'][i] = detector.getId()
+
+        pixelsToFieldAngle = detector.getTransform(detector.makeCameraSys(PIXELS),
+                                                   detector.makeCameraSys(FIELD_ANGLE))
+        wcs = afwGeom.makeSkyWcs(pixelsToFieldAngle, orientation, flipX, boresight)
```

Comment:
Is there a reason you can't use `obs.base.utils.createInitialSkyWcs` here instead? Or just use the stack-generated WCS instead of creating your own?